improve aggregate performance by verifying the aggregate signature first

conradoplg opened the pull request:
Implements https://github.com/cfrg/draft-irtf-cfrg-frost/pull/360

upbqdn commented on frost-core/src/frost.rs:
```diff
@@ -396,8 +366,53 @@ where
+        .group_public
+        .verify(signing_package.message(), &signature);
+
+    // If verification failed; verify each share to find the cheater.
```
The "only if" construct might be handy here:
```suggestion
    // Only if the verification of the aggregate signature failed; verify each share to find the cheater.
```

upbqdn reviewed (changes requested):
PR looks great. I noticed an outdated comment that I think we should change before merging this PR, so I'm requesting changes.

See https://github.com/ZcashFoundation/frost/blob/78e67d175b71deae26dec6ab608918e1e55b0234/frost-core/src/frost.rs#L326-L327

dconnolly commented on frost-core/src/frost.rs:
```diff
@@ -396,8 +366,53 @@ where
+        .verify(signing_package.message(), &signature);
+
+    // If verification failed; verify each share to find the cheater.
+    // This approach is more efficient since we don't need to verificate all shares
```
```suggestion
    // This approach is more efficient since we don't need to verify all shares
```

dconnolly reviewed (dismissed):
🎉

upbqdn reviewed (commented):
Looks great. However, I still think we should update this comment. https://github.com/ZcashFoundation/frost/blob/78e67d175b71deae26dec6ab608918e1e55b0234/frost-core/src/frost.rs#L326-L327

Unfortunately, GitHub won't let me make a suggestion.

conradoplg commented:
> Looks great. However, I still think we should update this comment.
> 
> https://github.com/ZcashFoundation/frost/blob/78e67d175b71deae26dec6ab608918e1e55b0234/frost-core/src/frost.rs#L326-L327
> 
> Unfortunately, GitHub won't let me make a suggestion.

Good catch, fixed it in a904af6, I also removed some outdated wording
